tests/HttpClientInterception.Tests/Examples.cs

```diff
@@ -520,5 +517,113 @@ public static class Examples
+                .ForHttp()
+                .ForAnyHost()
+                .ForPath("orders")
+                .ForQuery("id=12")
+                .WithStatus(HttpStatusCode.OK)
+                .WithContent(expected);
+
+            var options = new HttpClientInterceptorOptions().Register(builder);
+
+            using (var client = options.CreateHttpClient())
+            {
+                // Act
+                actual = await client.GetStringAsync("http://myhost.net/orders?id=12");
+            }
+
+            // Assert
+            actual.ShouldBe(expected);
+
+            using (var client = options.CreateHttpClient())
+            {
+                // Act
+                actual = await client.GetStringAsync("http://myotherhost.net/orders?id=12");
+            }
+
+            // Assert
+            actual.ShouldBe(expected);
+        }
+
+        [Fact]
+        public static async Task Use_Default_Response_For_Unmatched_Requests()
+        {
+            // Arrange
+            var options = new HttpClientInterceptorOptions()
+            {
+                OnMissingRegistration = (request) => Task.FromResult(new HttpResponseMessage(HttpStatusCode.NotFound))
+            };
+
+            using (var client = options.CreateHttpClient())
+            {
+                // Act
+                using (var response = await client.GetAsync("https://google.com/"))
+                {
+                    // Assert
+                    response.StatusCode.ShouldBe(HttpStatusCode.NotFound);
+                }
+            }
+        }
+
+        [Fact]
+        public static async Task Use_Custom_Request_Matching()
+        {
+            // Arrange
+            var builder = new HttpRequestInterceptionBuilder()
+                .Requests().For((request) => request.RequestUri.Host == "google.com")
+                .Responds().WithContent(@"<!DOCTYPE html><html dir=""ltr"" lang=""en""><head><title>Google Search</title></head></html>");
+
+            var options = new HttpClientInterceptorOptions().Register(builder);
+
+            using (var client = options.CreateHttpClient())
+            {
+                // Act and Assert
+                (await client.GetStringAsync("https://google.com/")).ShouldContain("Google Search");
+                (await client.GetStringAsync("https://google.com/search")).ShouldContain("Google Search");
+                (await client.GetStringAsync("https://google.com/search?q=foo")).ShouldContain("Google Search");
+            }
+        }
+
+        [Fact]
+        public static async Task Use_Custom_Request_Matching_With_Priorities()
+        {
+            // Arrange
+            var options = new HttpClientInterceptorOptions()
+            {
+                ThrowOnMissingRegistration = true,
+            };
+
+            var builder = new HttpRequestInterceptionBuilder()
+                .Requests().For((request) => request.RequestUri.Host == "google.com").HavingPriority(1)
```

Comment:
Would this test pass even without the concept of priority since they are registered in the order of their heirarchy?

Reply:
I did consider re-ordering this to prove that, but as it was an example I went for readability. I'll add another unit test to another class that proves the order is not important.